internal/resolver/dns/dns_resolver.go

```diff
@@ -201,18 +207,49 @@ func (d *dnsResolver) Close() {
 
 func (d *dnsResolver) watcher() {
 	defer d.wg.Done()
+	backoffIndex := 1
+	nextBackoffRate := backoff.DefaultExponential.Backoff(backoffIndex)
 	for {
 		select {
 		case <-d.ctx.Done():
+			d.pollTimer.Stop()
```

Comment:
After initializing, you can do `defer pollTimer.Stop()` if you make it a local, then you don't have to remember to stop it.

Reply:
Done.